internal/encoding/yaml/codec.go

```diff
@@ -1,12 +1,20 @@
 package yaml
 
-import "gopkg.in/yaml.v2"
+import (
+	"bytes"
+
+	"gopkg.in/yaml.v3"
+)
 
 // Codec implements the encoding.Encoder and encoding.Decoder interfaces for YAML encoding.
 type Codec struct{}
 
 func (Codec) Encode(v interface{}) ([]byte, error) {
-	return yaml.Marshal(v)
+	var buf bytes.Buffer
+	encoder := yaml.NewEncoder(&buf)
+	encoder.SetIndent(1)
```

Comment:
What's the default value for this? Would it make sense to use it instead?

Reply:
@sagikazarmark I created a new PR using these defaults & rebased on master: #1264